docs/reference/mapping/types/wildcard.asciidoc

```diff
@@ -0,0 +1,52 @@
+[role="xpack"]
+[testenv="basic"]
+[[wildcard]]
+=== Wildcard datatype
+++++
+<titleabbrev>Wildcard</titleabbrev>
+++++
+
+A `wildcard` field stores values optimised for wildcard queries.
+Wildcard queries are possible on other field types but suffer from constraints:
+* `text` fields limit matching of any wildcard expressions to individual tokens rather than the original whole value held in a field
+* `keyword` fields are untokenized but slow at performing wildcard queries (especially patterns with leading wildcards).
+
+Internally the `wildcard` field indexes the whole field value using ngrams and stores the full string in compressed chunks.
+The index is used as a rough filter to cut down the number of values that are then checked by retrieving and checking the full values from the compressed store.
+Storage costs are typically lower than those of `keyword` fields
```

Comment:
Maybe leave the size reduction point in but mention the downside of slower exact-match full-term queries? Critics of this field assumed it would cost more in disk space than keyword.

Reply:
ok, let's do that